lib/util/http/github.spec.ts

```diff
@@ -77,6 +77,32 @@ describe('util/http/github', () => {
       expect(req.headers.authorization).toBe('token 123test');
     });
 
+    it('supports fetching asset with octet-stream header', async () => {
+      const content = 'foobar';
+      hostRules.add({ hostType: 'github', token: 'x-access-token:123test' });
+      httpMock
+        .scope(githubApiHost)
+        .get('/repos/some-owner/some-repo/releases/tags/some-tag')
+        .reply(200, Buffer.from(content, 'utf-8'), {
+          'content-type': 'application/octet-stream',
+          'content-length': `${content.length}`,
+          'content-disposition': 'attachment; filename=reply_file_2.tar.gz',
+        });
+
+      const res = await githubApi.get(
+        '/repos/some-owner/some-repo/releases/tags/some-tag',
+        {
+          headers: { accept: 'application/octet-stream' },
+        }
+      );
```

Comment:
This looks wrong, as `Http.get` returns string type. I don't like this, as we again have wrong runtime types.

Can't you use `Http.stream()` for binary downloads instead? So you can directly pipe it to a files, like we do for terraform binaries hash computation ?

https://github.com/renovatebot/renovate/blob/f5f06c73be8c5b29f3458c40bc0d18259685296e/lib/modules/manager/terraform/lockfile/hash.ts#L75-L79

or bazel hash

https://github.com/renovatebot/renovate/blob/10302eba68fba11b8e6aa667529d64d708a54d97/lib/modules/manager/bazel/update.ts#L59-L61